snuba/query/query_processor.py

```diff
@@ -40,11 +46,59 @@ class ExtensionQueryProcessor(QueryProcessor[ExtensionData]):
     """
 
     @abstractmethod
-    def process_query(self, query: Query, extension_data: ExtensionData) -> None:
+    def process_query(
+            self, query: Query,
+            extension_data: ExtensionData,
+            request_settings: Mapping[str, bool],
+            query_hints: Mapping[str, Any],
+            stats: Mapping[str, Any],
+    ) -> None:
         raise NotImplementedError
 
 
 class DummyExtensionProcessor(ExtensionQueryProcessor):
 
-    def process_query(self, query: Query, extension_data: ExtensionData) -> None:
+    def process_query(
+            self,
+            query: Query,
+            extension_data: ExtensionData,
+            request_settings: Mapping[str, bool],
+            query_hints: Mapping[str, Any],
+            stats: Mapping[str, Any],
+    ) -> None:
         return query
+
+
+class ProjectExtensionProcessor(ExtensionQueryProcessor):
+
+    def process_query(
+            self,
+            query: Query,
+            extension_data: ExtensionData,
+            request_settings: Mapping[str, bool],
+            query_hints: Mapping[str, Any],
+            stats: Mapping[str, Any],
```

Comment:
I feel it may be clearer if `process_query` returned `query_hints` and `stats` (and `query` but that's complicated) and then we fold on them at the call site. But that feels like a very functional way to solve the problem, which doesn't seem consistent with the rest of the codebase.

Curious to hear other opinions.